src/main/scala/rocket/Frontend.scala

```diff
@@ -59,15 +59,15 @@ class FrontendIO(implicit p: Parameters) extends CoreBundle()(p) {
   val perf = new FrontendPerfEvents().asInput
 }
 
-class Frontend(val icacheParams: ICacheParams, hartid: Int)(implicit p: Parameters) extends LazyModule {
+class Frontend(val icacheParams: ICacheParams, staticIdForMetadata: Int)(implicit p: Parameters) extends LazyModule {
   lazy val module = new FrontendModule(this)
-  val icache = LazyModule(new ICache(icacheParams, hartid))
+  val icache = LazyModule(new ICache(icacheParams, staticIdForMetadata))
   val masterNode = icache.masterNode
   val slaveNode = icache.slaveNode
+  val resetVectorSinkNode = BundleBridgeSink[UInt](Some(() => Output(UInt(masterNode.edges.out.head.bundle.addressBits.W))))
```

Comment:
i guess i need to read up on what BundleBridge parameters take these days... why does the Sink take  a function that generates an Output?

Reply:
Ah, this is a bit of a kludge; BundleBridge is written assuming it is bridging an IO defined from the source/"master"'s perspective. In this case, the sink is defining the bundle to be bridged (because only sinks know how wide to make the bundle in this case), and the bridge makes an external IO at the source side by `Flipping` the bundle. But if you just give it an undirected `UInt` instead of some "IO" Bundle class with directionality, Chisel complains when doing the flip.

It's plausible I could build something smarter into the `BundleBridgeSource.makeIO` function, where it assumes it should make an input IO if the direction it was given is undefined? @jackkoenig any thoughts

Reply:
I don't entirely understand, clarifying questions:
1. Is `resetVectorSinkNode` still the sink or are you inverting the direction of the BundleBridge?
2. Is is just that the sink is defining the type and for that to work it has to give the `UInt` a direction?

Reply:
1. It is still the sink
2. Yes, the sink is defining the type, and the way the BundBridgeSource.makeIO is currently written, it `Flipped`s the bundle (because it was written assuming the source is defining a directionalized IO).

So I think more concretely my question is, does it seem sensible to treat an undirected bundle as something that needs to be flipped? I guess it is actually based on whether the source or the defined the type? :rubber_duck: I think I can probably do something sensible with sufficient application of `DataMirror.directionOf`

Reply:
@jackkoenig if you get a chance can you take a look at https://github.com/chipsalliance/rocket-chip/pull/2521/commits/9c10a2c8e8590b2d8ddd31a0e92f8cea07690a3d to see if it looks kosher

Reply:
Yeah I think that looks good. Yeah I'm not sure if there's something Chisel should be doing here to help you or if this is just the right way to do it but 9c10a2c LGTM!